flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_AOC_OFPData.js

```diff
@@ -209,10 +209,9 @@ class CDUAocOfpData {
         };
         mcdu.onRightInput[4] = () => {
             getSimBriefOfp(mcdu, updateView, () => {
-                setDefaultWeights(mcdu.simbrief.paxWeight, mcdu.simbrief.bagWeight).then(() => {
-                    getSimBriefOfp(mcdu, updateView, () => {
-                        setTargetPax(mcdu.simbrief.paxCount);
-                        setTargetCargo(mcdu.simbrief.bagCount, mcdu.simbrief.freight);
+                setDefaultWeights(mcdu.simbrief.paxWeight, mcdu.simbrief.bagWeight);
+                setTargetPax(mcdu.simbrief.paxCount).then(() => {
+                    setTargetCargo(mcdu.simbrief.bagCount, mcdu.simbrief.freight).then(() => {
                         updateView();
                     });
                 });
```

Comment:
Assuming these functions can be safely run in parallel, this could also be written as:

```suggestion
                Promise.all([
                    setTargetPax(mcdu.simbrief.paxCount),
                    setTargetCargo(mcdu.simbrief.bagCount, mcdu.simbrief.freight)
                ]).then(() => {
                    updateView();
                });
```

Reply:
Just tested, cargo values need targetpax to have executed first to account for baggage weight. I'll refactor this down the line